algebraic_connectivity_notebook.ipynb

```diff
@@ -136,26 +135,27 @@
     "\n",
     "train_size = round(dataset_config[\"split\"] * len(dataset))\n",
     "train_dataset = dataset[:train_size]\n",
-    "test_dataset = dataset[train_size:]\n",
+    "if len(dataset) - train_size > 0:\n",
+    "    test_dataset = dataset[train_size:]\n",
+    "else:\n",
+    "    test_dataset = train_dataset\n",
+    "test_size = len(test_dataset)\n",
     "\n",
     "print()\n",
     "print(f\"Number of training graphs: {len(train_dataset)}\")\n",
     "print(f\"Number of test graphs: {len(test_dataset)}\")\n",
     "\n",
     "# Batch and load data.\n",
-    "# TODO: Batch size?\n",
-    "batch_size = dataset_config[\"batch_size\"] if dataset_config[\"batch_size\"] > 0 else len(train_dataset)\n",
+    "batch_size = int(np.ceil(dataset_config[\"batch_size\"] * len(train_dataset)))\n",
     "train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)\n",
-    "test_loader = DataLoader(test_dataset, batch_size=batch_size, shuffle=False)\n",
+    "test_loader = DataLoader(test_dataset, batch_size=min(test_size, batch_size), shuffle=False)\n",
     "\n",
-    "train_batch = None\n",
-    "test_batch = None\n",
     "# If the whole dataset fits in memory, we can use the following lines to get a single large batch.\n",
-    "#train_batch = next(iter(train_loader))\n",
-    "#test_batch = next(iter(test_loader))\n",
+    "train_batch = next(iter(train_loader))\n",
```

Comment:
Taking an item from the loader is one of the slowest operations in the training process. If we have only one batch, we can do this once and speed up training at least 2x.